entity-framework/core/what-is-new/ef-core-8.0/whatsnew.md

```diff
@@ -1127,3 +1127,468 @@ In addition to the enhancements described above, EF8 Preview 2 also includes som
+
+This translates to the following on SQL Server:
+
+```sql
+SELECT [p].[Name], (
+    SELECT TOP(1) CAST([d0].[value] AS date)
+    FROM OpenJson([p].[DaysVisited]) AS [d0]
+    ORDER BY CAST([d0].[value] AS date)) AS [FirstVisited], (
+    SELECT TOP(1) CAST([d1].[value] AS date)
+    FROM OpenJson([p].[DaysVisited]) AS [d1]
+    ORDER BY CAST([d1].[value] AS date) DESC) AS [LastVisited]
+FROM [Pubs] AS [p]
+ORDER BY (
+    SELECT TOP(1) CAST([d].[value] AS date)
+    FROM OpenJson([p].[DaysVisited]) AS [d]
+    ORDER BY CAST([d].[value] AS date))
+```
+
+And finally, just how often do we end up visiting the closest pub when taking the dog for a walk? Let's find out:
+
+<!--
+        var walksWithADrink = await context.Walks.Select(
+            w => new
+            {
+                WalkName = w.Name,
+                PubName = w.ClosestPub.Name,
+                Count = w.DaysVisited.Count(v => w.ClosestPub.DaysVisited.Contains(v)),
+                TotalCount = w.DaysVisited.Count
+            }).ToListAsync();
+-->
+[!code-csharp[WalksWithADrink](../../../../samples/core/Miscellaneous/NewInEFCore8/PrimitiveCollectionsSample.cs?name=WalksWithADrink)]
+
+This translates to the following on SQL Server:
+
+```sql
+SELECT [w].[Name] AS [WalkName], [p].[Name] AS [PubName], (
+    SELECT COUNT(*)
+    FROM OpenJson([w].[DaysVisited]) AS [d]
+    WHERE EXISTS (
+        SELECT 1
+        FROM OpenJson([p].[DaysVisited]) AS [d0]
+        WHERE CAST([d0].[value] AS date) = CAST([d].[value] AS date) OR ([d0].[value] IS NULL AND [d].[value] IS NULL))) AS [Count], (
+    SELECT COUNT(*)
+    FROM OpenJson([w].[DaysVisited]) AS [d1]) AS [TotalCount]
+FROM [Walks] AS [w]
+INNER JOIN [Pubs] AS [p] ON [w].[ClosestPubId] = [p].[Id]
+```
+
+And reveals the following data:
+
+```none
+The Prince of Wales Feathers was visited 5 times in 8 "Ailsworth to Nene" walks.
+The Prince of Wales Feathers was visited 6 times in 9 "Caster Hanglands" walks.
+The Royal Oak was visited 6 times in 8 "Ferry Meadows" walks.
+The White Swan was visited 7 times in 9 "Woodnewton" walks.
+The Eltisley was visited 6 times in 8 "Eltisley" walks.
+Farr Bay Inn was visited 7 times in 11 "Farr Beach" walks.
+Farr Bay Inn was visited 7 times in 9 "Newlands" walks.
+```
+
+Looks like beer and dog walking are a winning combination!
+
+#### Primitive collections in JSON documents
+
+In all the examples above, column for primitive collection contains JSON. However, this is not the same as mapping [an owned entity type to a column containing a JSON document](xref:core/what-is-new/ef-core-7#json-columns), which was introduced in EF7. But what if that JSON document itself contains a primitive collection? Well, all the queries above still work in the same way! For example, imagine we move the _days visited_ data into an owned type `Visits` mapped to a JSON document:
```

Comment:
Yeah, the below works but we don't really support this properly 😬 Since the primitive array is encoded as a string within the owned JSON document, some stuff doesn't work correctly. It may be worth at least mentioning this problem (and maybe even leave this section out until we fix this)

Reply:
Will merge for now, and we can discuss. I was impressed that all the queries I tried actually worked. Hopefully nobody looks at the actual JSON. ;-)